src/validation/document_validator.py

```diff
@@ -24,10 +24,13 @@
 from src.validation.validation_message import ValidationMessage, ValidationContext, SpdxElementType
 
 
-def validate_full_spdx_document(document: Document, spdx_version: str) -> List[ValidationMessage]:
+def validate_full_spdx_document(document: Document, spdx_version: str = None) -> List[ValidationMessage]:
     validation_messages: List[ValidationMessage] = []
 
-    validation_messages.extend(validate_creation_info(document.creation_info))
+    if not spdx_version:
+        spdx_version = document.creation_info.spdx_version
```

Comment:
instead of setting the version here, I think it would be cleaner to make the version optional in the `validation_messages.extend(validate_creation_info(` call and only assert on it, if it is set.

Reply:
I agree that it should be optional in `validate_creation_info()`, but we have to set the version here, nevertheless, as it is a mandatory field for other validations like `validate_relationships()` (and more to come later).

Reply:
you could still move the spdx version validation up, to have it cleaner

Reply:
Its just a minor improvement, but:
```suggestion
    validation_messages.extend(validate_creation_info(document.creation_info, spdx_version))
    
    if not spdx_version:
        spdx_version = document.creation_info.spdx_version
```

Reply:
I have extracted the version validation to be handled before everything else, now.